### PR TITLE
Add destroy() function

### PR DESCRIPTION
**Pull request by facekapow**
This PR allows for `http`'s `res.on('end')` to work. Should I add a test? To ensure it works?

EDIT: @kesla I added tests that should pass, they work for me, at least.

### CODE REVIEW

File: lib/socket.js [hunk 1]
```diff
@@ -44,6 +44,11 @@ class Socket extends Duplex {
   setTimeout () {}
 
   destroySoon () {}
+  
+  destroy() {
+    this._socket._destroy();
+    this.writable = false;
```
Comment (iefserge) [inline lib/socket.js:54]: Nice! `this.writable = false` fixes assertion error I have.

File: lib/socket.js [hunk 2]
```diff
@@ -44,6 +44,11 @@ class Socket extends Duplex {
   setTimeout () {}
 
   destroySoon () {}
+  
+  destroy() {
+    this._socket._destroy();
```
Comment (iefserge) [inline lib/socket.js]: Noo, we shouldn't call private socket methods

File: lib/socket.js (hunk 1 above)
Comment (facekapow) [inline lib/socket.js:54]: Yeah, I was getting the AssertionError, too. I traced it to `assert(!socket.writable)`. I just set `writable` to false to resolve it.

File: lib/socket.js (hunk 2 above)
Comment (facekapow) [inline lib/socket.js]: So, use `close()` instead?

Comment (iefserge) [inline lib/socket.js]: yeah, probably `close()`, it should just close the connection in this case

Comment (facekapow) [inline lib/socket.js]: Fixed that.

**Comment by facekapow** [comment]
@iefserge @kesla Does this look ok?

**Comment by kesla** [comment]
@ArielAbreu Why were the tests removed?

**Comment by facekapow** [comment]
@kesla They were producing some errors that I didn't have time to figure out. I can fix the errors tommorrow and add the tests back in. So, until tommorrow, they're not going to be in there.

**Comment by facekapow** [comment]
@kesla I fixed the error. It was giving me an `AssertionError` because `destroySoon()` wasn't making the socket non-writable like `destroy()`. I fixed it by making `destroySoon()` do the same as `destroySoon()` (except for closing the socket, that's `destroy()`'s job), I also do it in `onend`, better safe than sorry! I was able to add the tests back in, and they pass without error!

File: test/runner.js [hunk 3]
```diff
@@ -1,5 +1,3 @@
-'use strict';
-
```
Comment (kesla) [inline test/runner.js:2]: why removing `use strict`?

File: test/runner.js [hunk 4]
```diff
@@ -32,7 +30,7 @@ runtimeify({
   }
 
   const qemu = cp.spawn(qemuPath,
-    [ `${__dirname}/initrd`, '--verbose', '--nographic' ]
+    [ __dirname + '/initrd', '--verbose', '--nographic' ]
```
Comment (kesla) [inline test/runner.js]: Oh, you're not on io.js, right?

File: lib/socket.js [hunk 5]
```diff
@@ -23,6 +23,8 @@ class Socket extends Duplex {
 
     this._socket.onend = function () {
       self.push(null);
+      self.writable = false;
+      self.readable = false;
```
Comment (kesla) [inline lib/socket.js]: let's call destroySoon here!

File: test/runner.js (hunk 3 above)
Comment (facekapow) [inline test/runner.js:2]: It tells me 'use of const in strict mode'.

File: test/runner.js (hunk 4 above)
Comment (facekapow) [inline test/runner.js]: No, im on Node 0.12, but I feel like i'm missing out on ES6 without io.js, i'm going to install it and re-add ecerything the way it was before.

File: lib/socket.js (hunk 5 above)
Comment (facekapow) [inline lib/socket.js]: Ok, i'll fix that.

**Comment by facekapow** [comment]
@kesla Now i'm using `io.js` (ES6 == way better), so I made `runner.js` the way it used to be, and now `this._socket.onend` calls `self.destroySoon()`.

**Comment by kesla** [comment]
awesome, thanks a lot!

**Comment by facekapow** [comment]
No problem!

**Pull request merged**